src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala

```diff
@@ -300,7 +300,11 @@ case class ErgoTransaction(override val inputs: IndexedSeq[Input],
           log.debug(s"Reemission tokens to burn: $toBurn")
           val reemissionOutputs = outputCandidates.filter { out =>
             require(!out.tokens.contains(reemissionTokenId), "outputs contain reemission token")
-            out.ergoTree == payToReemissionContract
+            if (chainSettings.isMainnet) {
+              out.ergoTree == payToReemissionContract
+            } else {
+              out.ergoTree.toProposition(true) == payToReemissionContract.toProposition(true)
+            }
```

Comment:
could you please describe, why the original comparison `out.ergoTree == payToReemissionContract` won't affect mainnet?
Also, shoudn't testnet also have protocol upgrades to be on par with mainnet?

Reply:
The testnet was activated with old clients, issuing and accepting (the most important part) v0 transactions burning re-emission tokens while sending ERGs to pay-to-reemission outputs. Such transactions appeared in block #206,155. Mainnet has no such transactions in the mainnet (and they could be problematic after EIP-27 activation only).

To be on par, we need for testnet relaunch at this point.